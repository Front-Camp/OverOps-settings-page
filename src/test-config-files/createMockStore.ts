import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middleware = [
  thunk
];

const createMockStore = () => configureMockStore(middleware);

export default createMockStore;
