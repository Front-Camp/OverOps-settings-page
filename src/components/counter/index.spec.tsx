import * as React from 'react';
import { shallow } from 'enzyme';

import Counter from './index';
import createMockStore from '../../test-config-files/createMockStore';

describe('Components', () => {
  const mockStore = createMockStore();
  const store = mockStore({});

  test('Counter', () => {
    // @ts-ignore
    const checkbox = shallow(<Counter store={store} />);

    // Snapshot demo
    expect(checkbox).toMatchSnapshot();
  });
});
