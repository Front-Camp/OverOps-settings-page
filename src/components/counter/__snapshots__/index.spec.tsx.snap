// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components CheckboxWithLabel changes the text after click 1`] = `
<Counter
  actions={
    Object {
      "decrement": [Function],
      "increment": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Components Counter 1`] = `
<Counter
  actions={
    Object {
      "decrement": [Function],
      "increment": [Function],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
